satellite/satellitedb/containment.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type containment struct {
+	db *dbx.DB
+}
+
+// Get gets the pending audit by node id
+func (containment *containment) Get(ctx context.Context, id pb.NodeID) (*audit.PendingAudit, error) {
+	if id.IsZero() {
+		return nil, audit.ContainError.New("node ID empty")
+	}
+
+	pending, err := containment.db.Get_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return nil, audit.ErrContainedNotFound.New(id.String())
+	}
+	if err != nil {
+		return nil, audit.ContainError.Wrap(err)
+	}
+
+	return convertDBPending(pending)
+}
+
+// IncrementPending will create a new pending audit entry, or increase its reverify count if it already exists
+func (containment *containment) IncrementPending(ctx context.Context, pendingAudit *audit.PendingAudit) error {
+	statement := containment.db.Rebind(
+		`INSERT INTO pending_audits (node_id, piece_id, stripe_index, share_size, expected_share_hash, reverify_count)
+		VALUES (?, ?, ?, ?, ?, ?)
+		ON CONFLICT(node_id)
+		DO UPDATE SET reverify_count = pending_audits.reverify_count + 1`,
+	)
+	_, err := containment.db.ExecContext(ctx, statement,
+		pendingAudit.NodeID.Bytes(), pendingAudit.PieceID.Bytes(), pendingAudit.StripeIndex, pendingAudit.ShareSize, pendingAudit.ExpectedShareHash, pendingAudit.ReverifyCount,
+	)
+	if err != nil {
+		return audit.ContainError.Wrap(err)
+	}
+	return nil
+}
+
+// Delete deletes the pending audit
+func (containment *containment) Delete(ctx context.Context, id pb.NodeID) error {
+	if id.IsZero() {
+		return audit.ContainError.New("node ID empty")
+	}
+	isDeleted, err := containment.db.Delete_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return audit.ErrContainedNotFound.New(id.String())
```

Comment:
Do we really want to return an error in this case? I think this method should just delete everything about the given node from the containment db and not return an error in case there isn't anything about this node.

Also, I think if there is nothing to delete, the `Delete_PendingAudits_By_NodeId` will return `isDeleted == false` instead of this error, but I could be wrong.

Reply:
okay, I can check this with a test. Should I not log anything either?

Reply:
Just check if in this cases that  `Delete_PendingAudits_By_NodeId` will indeed return `isDeleted == false` instead of `sql.ErrNoRows`. In such case you should just remove this if-clause.

Reply:
yep it will return `isDeleted == false`. so I'll leave the `if !isDeleted` check and remove the sql.ErrNoRows one.

Reply:
or should I remove both so that no error is returned if there's nothing to delete?

Reply:
oh, just saw your other comment thread- I'll do that. sorry for the triple reply.

Reply:
What @kaloyan-raev wrote looks good because if this could be call concurrently then, non-returning an error if it doesn't exist will make the usage easier.